File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        
        for i in range(len(arr)-1):
            temp = []
            for x in range(i+1,len(arr)):
                temp.append(arr[x])
                temp.sort()
            arr[i] = temp.pop()
        arr[len(arr)-1] = -1
        return arr
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> freq = new HashMap<>();
        for(int n : nums){
            freq.put(n,freq.getOrDefault(n,0)+1);
        }
        Map<Integer,List<Integer>> bucket = new HashMap<>();
        List<Integer> unqs = new ArrayList<>(freq.keySet());
        for(int unq : unqs){
            int temp = freq.get(unq);
            bucket.put(temp,new ArrayList<>());
        }
        for(int unq : unqs){
            bucket.get(freq.get(unq)).add(unq);
        }
        int[] res = new int[k];
        List<Integer> reversed = new ArrayList<>(bucket.keySet());
        Collections.sort(reversed);
        Collections.reverse(reversed);
        int idx = 0;
        for(int f : reversed){
            for(int num : bucket.get(f)){
                res[idx++] = num;
                if(idx == k)    return res;
            }
        }
        return res;

    }
}
